src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -726,6 +726,61 @@ theorem getLsb_sshiftRight (x : BitVec w) (s i : Nat) :
         Nat.not_lt, decide_eq_true_eq]
       omega
 
+/-! ### signExtend -/
+
+/-- Equation theorem for 'Int.sub' when both arguments are 'negSucc' -/
+private theorem Int.ofNat_sub_ofNat_of_lt {n m : Nat} (hlt : n < m) :
+    (n : Int) - (m : Int) = -(↑(m - 1 - n) + 1) := by
+  omega
+
+/-- Equation theorem for 'Int.mod' -/
+private theorem Int.negSucc_emod (m : Nat) (n : Int) :
+    -(m + 1) % n = Int.subNatNat (Int.natAbs n) ((m % Int.natAbs n) + 1) := rfl
+
+
+/-- To sign extend when the msb is false, then sign extension is the same as truncation -/
+theorem signExtend_eq_neg_truncate_neg_of_msb_false {x : BitVec w} {v : Nat} (hmsb : x.msb = false) :
+    (x.signExtend v) = x.zeroExtend v := by
+  ext i
+  by_cases hv : i < v
+  · simp only [getLsb_zeroExtend, hv, decide_True, Bool.true_and]
+    rw [signExtend, getLsb]
+    simp only [toNat_ofInt]
+    rw [BitVec.toInt_eq_msb_cond]
+    simp only [hmsb, Bool.false_eq_true, ↓reduceIte]
+    rw [Int.ofNat_mod_ofNat, Int.toNat_ofNat]
+    simp only [Nat.testBit_mod_two_pow, hv, decide_True, Bool.true_and]
+    rw [BitVec.testBit_toNat]
```

Comment:
It seems a lot of switching between `simp only` and `rw`, but I guess some is necessary. Not sure if 
```
  · simp only [signExtend, getLsb, getLsb_zeroExtend, hv, decide_True, Bool.true_and, toNat_ofInt, 
      BitVec.toInt_eq_msb_cond, hmsb, ↓reduceIte]
    rw [Int.ofNat_mod_ofNat, Int.toNat_ofNat, Nat.testBit_mod_two_pow]
    simp [BitVec.testBit_toNat]
```
is an improvement or not. Up to you.